cmake/BuildTesseract.cmake

```diff
@@ -0,0 +1,48 @@
+include(FetchContent)
+
+set(TESSERACT_SOURCE_URL
+    ""
+    CACHE STRING "URL of a Tesseract source tarball")
+
+set(TESSERACT_SOURCE_HASH
+    ""
+    CACHE STRING "Hash of a Tesseract source tarball")
+
+set(LEPTONICA_SOURCE_URL
+    ""
+    CACHE STRING "URL of a Leptonica source tarball")
+
+set(LEPTONICA_SOURCE_HASH
+    ""
+    CACHE STRING "Hash of a Leptonica source tarball")
+
+FetchContent_Declare(
+  tesseract_source
+  URL ${TESSERACT_SOURCE_URL}
+  URL_HASH ${TESSERACT_SOURCE_HASH})
+FetchContent_Populate(tesseract_source)
+
+FetchContent_Declare(
+  leptonica_source
+  URL ${LEPTONICA_SOURCE_URL}
+  URL_HASH ${LEPTONICA_SOURCE_HASH})
+FetchContent_Populate(leptonica_source)
+
+set(BUILD_TESSERACT_BASEDIR ${CMAKE_BINARY_DIR}/build-obs-ocr-dep)
+
+set(BUILD_TESSERACT_OUTPUTS ${BUILD_TESSERACT_BASEDIR}/release/tesseract/${CMAKE_BUILD_TYPE}/lib/libtesseract.a
+                            ${BUILD_TESSERACT_BASEDIR}/release/tesseract/${CMAKE_BUILD_TYPE}/lib/libleptonica.a)
+
+set(BUILD_TESSERACT_INCLUDE_DIR ${BUILD_TESSERACT_BASEDIR}/release/tesseract/${CMAKE_BUILD_TYPE}/include)
+
+add_custom_command(
+  OUTPUT ${BUILD_TESSERACT_OUTPUTS}
+  COMMAND ${CMAKE_SOURCE_DIR}/vendor/obs-ocr-deps/build-linux.sh ${CMAKE_BUILD_TYPE} main main
```

Comment:
maybe since this is linux only we check for it above and throw an error message if not on linux?

Reply:
Implemented